### PR TITLE
refactor for typescript

### PR DESCRIPTION
**Pull request by tbo47**
a more modern javascript syntax with compatibility with typescript

#### **Verify that...**

- [x] `npm test` passes
- [x] `npm run api` passes
- [x] `npm run lint` passes
- [x] `npm run build` still works

### CODE REVIEW

File: src/og/LonLat.js [hunk 1]
```diff
@@ -20,31 +20,33 @@ const INV_PI_BY_180_HALF_PI = INV_PI_BY_180 * HALF_PI;
...
 export class LonLat {
+    static SW_MERC; // LatLng
+    static NE_MERC; // LatLng
+    static NW_MERC_DEG; // LatLng
+    static SE_MERC_DEG; // LatLng
+
+    /**
+     * Longitude.
+     */
+    lon = 0;
```
Comment (Zemledelec) [inline src/og/LonLat.js:31]: in this case lon,lat, height are public.

File: src/og/Extent.js [hunk 2]
```diff
@@ -15,6 +15,8 @@ import { LonLat } from "./LonLat.js";
  * @param {LonLat} [ne] - North East extent corner coordiantes.
  */
 export class Extent {
+    southWest;
```
Comment (Zemledelec) [inline src/og/Extent.js:18]: public

**Review by Zemledelec — CHANGES_REQUESTED** [review]
As far as I understand we do not pull public variables before constructor? Right?

File: src/og/LonLat.js (hunk 1 above)
Comment (tbo47) [reply src/og/LonLat.js:31]: yes, they are public here. It's only if you prefix with `#` that they are private

**Comment by tbo47** [comment]
> As far as I understand we do not pull public variables before constructor? Right?

you don't need to have the public variables before the constructor in javascript but you need it for typescript

**Pull request closed without merging**